source/nodejs/adaptivecards/src/carousel.ts

```diff
@@ -419,6 +452,32 @@ export class Carousel extends Container {
     }
 }
 
+export class CarouselEvent {
+    type : Enums.CarouselEventType;
+    carouselId : string | undefined;
+    activeCarouselPageId : string | undefined;
+    activeCarouselPageIndex : number | undefined;
+    constructor(carousel : Carousel) {
+        this.carouselId = carousel.id;
+        this.activeCarouselPageIndex = carousel.currentPageIndex;
+        this.activeCarouselPageId = (this.activeCarouselPageIndex != undefined) ? carousel.getItemAt(this.activeCarouselPageIndex).id : undefined; 
+        this.type = carousel.currentEventType;
+    }
+}
+
+function raiseCarouselEvent(carousel: Carousel, eventType : Enums.CarouselEventType) {
+    const card = carousel.getRootElement() as AdaptiveCard;
+    const onCarouselEventHandler =
+        card && card.onCarouselEvent
+            ? card.onCarouselEvent
+            : AdaptiveCard.onCarouselEvent;
+
+    if (onCarouselEventHandler && eventType == Enums.CarouselEventType.PaginationInteractionType) {
+        onCarouselEventHandler(new CarouselEvent(carousel));
```

Comment:
Why do we only call this for `PaginationInteractionType`?

Reply:
PaginationInteractionType will be get called when slide transition end event is fired.
the actual event type is one saved in the previous event that caused the transition.